src/engine/strat_engine/setup.rs

```diff
@@ -98,20 +98,25 @@ pub fn find_all() -> EngineResult<HashMap<PoolUuid, Vec<PathBuf>>> {
 /// Returns None if no metadata found for this pool.
 pub fn get_metadata(pool_uuid: PoolUuid, devnodes: &[PathBuf]) -> EngineResult<Option<PoolSave>> {
 
-    // Get pairs of device nodes and matching BDAs
-    // If no BDA, or BDA UUID does not match pool UUID, skip.
-    // If there is an error reading the BDA, error. There could have been
-    // vital information on that BDA, for example, it may have contained
-    // the newest metadata.
-    let mut bdas = Vec::new();
-    for devnode in devnodes {
-        let bda = try!(BDA::load(&mut try!(OpenOptions::new().read(true).open(devnode))));
-        if let Some(bda) = bda {
-            if *bda.pool_uuid() == pool_uuid {
-                bdas.push((devnode, bda));
-            }
-        }
-    }
+    // Get pairs of device nodes and matching BDAs from readable and
```

Comment:
I thought the semantics through very carefully when I wrote this initially, and I continue to believe that it is correct. If there is an error, then nothing is known about the BDA, and this method can not guarantee that it is returning the most recent metadata available from the list of blockdevs passed. Therefore it should return an error. If for all of the devices the BDA does not exist or it does not match the pool, then his method can confidently return None. There's a big distinction between "I don't know" and "I know there aren't any" and that's what this code was capturing.